### DISCUSSION TITLE
Tweaking "Import build" prompt

### DISCUSSION THREAD

**Discussion by SuperFola** [scalameta/nvim-metals#537] [Q&A] [answered]
Hi there!

I've been using nvim-metals for a while now and recently added noice to my configuration (basically changes the interface of neovim), and I was wondering if it was possible to tweak the "Import build" prompt (shown as a popup with notify through noice in my setup) so that I don't have to press return to validate my choice.

This is because the prompt is shown in a popup and there is no indication that you can still type in it/it's validated (a limitation of the api of notify/noice I guess)

![prompt popup](https://user-images.githubusercontent.com/10694085/216563200-79250f3d-8f53-4cd5-961d-2e856d794a05.png)

**Comment by ckipp01** [comment, marked as answer]
Hey @SuperFola, thanks for the question. I'm not super familiar with `noice`, but I'm a bit surprised that they don't handle this differently, unless I'm missing something. But behind the scenes the build import prompt is using a [`showMessageRequest`]( https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#window_showMessageRequest). By definition it expects to be able to do the following:

> In addition to the show message notification the request allows to pass actions and to wait for an answer from the client.

So this should be supported. Have you by chance checked with `noice` to see if maybe there is a setting or something that you might be missing? I'd be surprised if this wasn't supported.

**Reply by SuperFola** [reply]
Is there a way for me to track where / when this is called? I've looked at the source code of noice, nvim-notify and nvim-lspconfig but couldn't find any reference to `showMessageRequest` ; it works but I don't know how

**Reply by ckipp01** [reply]
Sure, there is a few ways, but one of the easiest ones would be to create a [`lsp.trace.json`](https://scalameta.org/metals/docs/contributors/getting-started#json-rpc-trace) file in your `.metals/` dir. Then you'll be able to see the full LSP communication between Metals and Neovim.
